rohmu/object_storage/azure.py

```diff
@@ -113,6 +113,42 @@ def __init__(
         self.container = self.get_or_create_container(self.container_name)
         self.log.debug("AzureTransfer initialized, %r", self.container_name)
 
+    @staticmethod
+    def conn_string(
+        account_name: str,
+        account_key: Optional[str],
+        azure_cloud: Optional[str],
+        host: Optional[str],
+        port: Optional[int],
+        is_secure: bool,
+    ) -> str:
+        if not host and not port:
+            try:
+                endpoint_suffix = ENDPOINT_SUFFIXES[azure_cloud]
+            except KeyError:
+                raise InvalidConfigurationError(f"Unknown azure cloud {repr(azure_cloud)}")
+
+            conn_str = (
+                "DefaultEndpointsProtocol=https;"
```

Comment:
Let's assume `is_secure == False`, but `host` and `port` are `None`. Returning connection string will be `https`, which is unexpected.

Reply:
Sure. Wasn't considering it for default endpoint, but there may be some scenario where this is needed. Refactored in last commit